itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/WarehouseInstance.java

```diff
@@ -394,16 +393,13 @@ WarehouseInstance verifyResults(String[] data) throws IOException {
   }
 
   WarehouseInstance verifyFailure(String[] data) throws IOException {
-    List<String> results = getOutput();
-    logger.info("Expecting {}", StringUtils.join(data, ","));
-    logger.info("Got {}", results);
-    boolean dataMatched = (data.length == results.size());
-    if (dataMatched) {
-      for (int i = 0; i < data.length; i++) {
-        dataMatched &= data[i].toLowerCase().equals(results.get(i).toLowerCase());
-      }
-    }
-    assertFalse(dataMatched);
+    final List<String> expectedResults =
+        Arrays.asList(data).stream().map(r -> r.toLowerCase()).collect(Collectors.toList());
+    final List<String> actualResults = getOutput().stream().map(r -> r.toLowerCase()).collect(Collectors.toList());
+
+    assertTrue("Data " + expectedResults + " should not be present in " + actualResults,
+        Collections.disjoint(expectedResults, actualResults));
+
```

Comment:
old and new code seem to be doing  different things
old is:
```
e[0] != r[0] || ... || e[n] != r[n]
```
new block is a condition which is checking a set operation between the elements...

why is this change necessary - how this is connected to a hadoop upgrade?